File: 0217-contains-duplicate/0217-contains-duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        int n = nums.size(), flag = 0;
            std::sort(nums.begin(),nums.end());
            for (int i = 0;i < n-1; i++)
            {
                if (nums[i] == nums[i + 1]) {
                    flag = 1;
                    break;    
                }
            }
        return flag;
    }
};

File: 0217-contains-duplicate/0217-contains-duplicate.rs
use std::collections::HashMap;

impl Solution {
    pub fn contains_duplicate(nums: Vec<i32>) -> bool {
        let mut hist = HashMap::new();
        
        for n in nums {
            if !hist.contains_key(&n) {
                hist.entry(n).and_modify(|n| *n += 1).or_insert(1);
            } else {
                return true;
            }
        }
        
        false
    }
}